src/main/java/org/joda/beans/ser/bin/AbstractBinReader.java

```diff
@@ -323,10 +325,7 @@ Object parseSimple(int typeByte, Class<?> type) throws Exception {
                 return Float.valueOf((float) value);
 
             } else {
-                if (value < Integer.MIN_VALUE || value > Integer.MAX_VALUE) {
-                    throw new IllegalArgumentException("Invalid binary data: Expected int, but was " + value);
-                }
-                return Integer.valueOf((int) value);
+                return Long.valueOf(value);
```

Comment:
I've made the change here too for consistency but the binary reader deduces the type correctly.